src/project-2.js

```diff
@@ -44,62 +79,104 @@ const isPrime = (num) => {
   // hint: a prime number is only evenly divisible by itself and 1
   // hint2: you can solve this using a for loop
   // note: 0 and 1 are NOT considered prime numbers
+  if (num === 0 || num === 1) {
+    return false;
+  }
+  if (num % 2 === 0 || num % 3 === 0) {
+    return false;
+  }
+  return true;
 };
 
 const returnFirst = (arr) => {
   // return the first item from the array
+  return arr[0];
 };
 
 const returnLast = (arr) => {
   // return the last item of the array
+  return arr[arr.length - 1];
 };
 
 const getArrayLength = (arr) => {
   // return the length of the array
+  return arr.length;
 };
 
 const incrementByOne = (arr) => {
   // arr is an array of integers
   // increase each integer by one
   // return the array
+  for (let i = 0; i < arr.length; i++) {
```

Comment:
So this will pass the tests, but this algorithm produces side effects to the array passed in. Not a big deal because it is a working solution, but do you know if there is a way to produce an algorithm that doesn't mutate the original array?

Reply:
We could pop each incremented value into a new array. All while being enclosed in a forEach which doesn't mutate the array in anyway in the MDN doc. 
We could also use map to achieve the results in a new array.

Reply:
Boom! Nice!